### PR TITLE
Remove last global variable

### LINKED ISSUE #1719: Remove last global variable
`transitMoyenInterconnexionsRecalculQuadratique`

### PR DESCRIPTION
**Pull request by payetvin**
close #1719

### CODE REVIEW

File: src/libs/antares/study/include/antares/study/runtime/runtime.h [hunk 1]
```diff
@@ -131,6 +131,8 @@ class StudyRuntimeInfos
     */
     bool quadraticOptimizationHasFailed;
```
Comment (guilpier-code) [inline src/libs/antares/study/include/antares/study/runtime/runtime.h:132]: Not directly related to this PR, but very easy to fix : 
Declaration : `void initializeThermalClustersInMustRunMode(Study& study) const;` 
is never defined, and a fortiori never called.
We should remove the declaration.

File: src/libs/antares/study/runtime/runtime.cpp [hunk 2]
```diff
@@ -343,6 +343,12 @@ bool StudyRuntimeInfos::loadFromStudy(Study& study)
     // Check if some clusters request TS generation
     checkThermalTSGeneration(study);
 
+    transitMoyenInterconnexionsRecalculQuadratique.resize(study.runtime.interconnectionsCount());
+    for (uint i = 0; i != study.runtime.interconnectionsCount(); i++)
+    {
+        transitMoyenInterconnexionsRecalculQuadratique[i].assign(HOURS_PER_YEAR, 0.);
+    }
```
Comment (guilpier-code) [inline src/libs/antares/study/runtime/runtime.cpp]: In block : 
```C++
transitMoyenInterconnexionsRecalculQuadratique.resize(study.runtime.interconnectionsCount());
for (uint i = 0; i != study.runtime.interconnectionsCount(); i++)
{
    transitMoyenInterconnexionsRecalculQuadratique[i].assign(HOURS_PER_YEAR, 0.);
}
```
when we call `study.runtime.interconnectionsCount()`, we're already in the context of **study.runtime**, because we are in class **StudyRuntimeInfos**.

Simple call to `interconnectionsCount()` is enough I guess.

File: src/solver/optimisation/opt_calcul_des_pmin_MUT_MDT.cpp [hunk 3]
```diff
@@ -23,7 +23,6 @@
 
 #include "antares/solver/optimisation/opt_fonctions.h"
```
Comment (guilpier-code) [inline src/solver/optimisation/opt_calcul_des_pmin_MUT_MDT.cpp:24]: ### Don't treat this remark, it'll be done in a separate PR 

Same as above : 
We should replace : 

```c++
#include "antares/solver/optimisation/opt_fonctions.h"
#include "antares/solver/optimisation/opt_structure_probleme_a_resoudre.h"
#include "antares/solver/simulation/sim_extern_variables_globales.h"
#include "antares/solver/simulation/sim_structure_donnees.h"
#include "antares/solver/simulation/simulation.h"
``` 
with ; 
```c++
#include "antares/solver/simulation/sim_structure_probleme_economique.h"
```

**Review by guilpier-code — CHANGES_REQUESTED** [review]

File: src/solver/variable/include/antares/solver/variable/economy/links/flowQuad.h [hunk 4]
```diff
@@ -268,6 +269,8 @@ class FlowQuad: public Variable::IVariable<FlowQuad<NextT>, NextT, VCardFlowQuad
 private:
     uint pLinkGlobalIndex;
     uint pNbHours;
+    std::vector<std::vector<double>>* transitMoyenInterco;
```
Comment (guilpier-code) [inline src/solver/variable/include/antares/solver/variable/economy/links/flowQuad.h]: About `std::vector<std::vector<double>>* transitMoyenInterco;`

Why not a reference to the vector (instead of a raw pointer) ?
It would avoid uses of symbol `&` and notation` (*transitMoyenInterco)`, always a bit harder to read.

Comment (guilpier-code) [inline src/solver/variable/include/antares/solver/variable/economy/links/flowQuad.h]: _Cosmetic_
About `std::vector<std::vector<double>>* transitMoyenInterco;`

Maybe renaming it into **transitMoyenInterco_** (trailing underscore) or **pTransitMoyenInterco** to tell reader it's a private data member.

**Review by guilpier-code — CHANGES_REQUESTED** [review]

File: src/solver/variable/include/antares/solver/variable/economy/links/flowQuad.h (hunk 4 above)
Comment (payetvin) [reply src/solver/variable/include/antares/solver/variable/economy/links/flowQuad.h]: Reference members must be initialized at construction, here we initialize it with another function

Comment (guilpier-code) [reply src/solver/variable/include/antares/solver/variable/economy/links/flowQuad.h]: True, I missed this point

Comment (guilpier-code) [reply src/solver/variable/include/antares/solver/variable/economy/links/flowQuad.h]: Maybe adding a **const** then : 
`const std::vector<std::vector<double>>* transitMoyenInterco;`

==> No, sorry, it'll force us to initialize the data member in the constructor. Forget that

**Review by guilpier-code — APPROVED** [review]
